sdk/storage/Azure.Storage.Blobs/src/AppendBlobClient.cs

```diff
@@ -369,7 +541,8 @@ protected sealed override BlobBaseClient WithCustomerProvidedKeyImpl(CustomerPro
             Metadata metadata,
             AppendBlobAccessConditions? accessConditions,
             bool async,
-            CancellationToken cancellationToken)
+            CancellationToken cancellationToken,
+            string operationName = Constants.Blob.Append.CreateOperationName)
```

Comment:
Nit - `cancellationToken` always goes last.  This would be best right above `bool async`

Reply:
Actually it gives me an error saying that optional parameters must appear after all required parameters

Reply:
Yes...  it would have to be difficult like that.   I suppose the other analysis rules for this won't fire because it's an internal method.  We could also use `ForwardsClientCalls`, but I don't feel strongly and you've already got this working.  Let's leave this for now.